### PR TITLE
Add tests for the New Vasp input sets

### PR DESCRIPTION
**Pull request by Zhuoying**
## Summary

Major changes:

- Follow #3484, add tests and clean up test_sets.py

## Todos

- Add more MD and NED-related tests
- Fix lints

## Checklist

- [ ] Google format doc strings added. Check with `ruff`.
- [ ] Type annotations included. Check with `mypy`.
- [ ] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

**Comment by Zhuoying** [comment]
Hi @utf, I have a question on `MITMDSet` and `MPMDSet`, is there any particular reason for skipping `ENCUT` in INCAR (set as None)?
https://github.com/materialsproject/pymatgen/blob/7e0c8f4ddea72367c2059865b0fa8cf8681663ea/pymatgen/io/vasp/sets.py#L2409
https://github.com/materialsproject/pymatgen/blob/7e0c8f4ddea72367c2059865b0fa8cf8681663ea/pymatgen/io/vasp/sets.py#L2482

The default settings in `MVLNPTMDSet` look more reasonable to me if someone just wants to manually run an MD wo workflow and grab some snapshots from a single run.
https://github.com/materialsproject/pymatgen/blob/7e0c8f4ddea72367c2059865b0fa8cf8681663ea/pymatgen/io/vasp/sets.py#L2559

**Comment by utf** [comment]
@Zhuoying, I didn't change the behaviour of the input sets. I just updated the implementation. E.g., for the two input sets you mentioned, see the old implementation where ENCUT was also removed from the input set.

- `MITMDSet`: https://github.com/materialsproject/pymatgen/blob/36e289cfb73572f54aef5528333673a4d969279b/pymatgen/io/vasp/sets.py#L2602
- `MPMDSet`: https://github.com/materialsproject/pymatgen/blob/36e289cfb73572f54aef5528333673a4d969279b/pymatgen/io/vasp/sets.py#L2691

I would highlight that the point of these tests is to ensure that the output of the new input set implementation is the same as the output of the old implementations.

**Comment by janosh** [comment]
thanks @Zhuoying! more tests are always appreciated! is this ready for review?

**Comment by Zhuoying** [comment]
@janosh Yeah, it is ready to review now. Pytest code coverage for `pymatgen/io/vasp/sets.py ` improved from 93% to 95%.
The two failing checks are not related to this PR. Please let me know if you have any comments.
For other `io.vasp` modules (`inputs, outputs, optics`), I may submit new PRs for them later if needed. 
Thanks.
```
Name                           Stmts   Miss  Cover
--------------------------------------------------
pymatgen/io/vasp/__init__.py       3      0   100%
pymatgen/io/vasp/help.py          35     35     0%
pymatgen/io/vasp/inputs.py      1071    109    90%
pymatgen/io/vasp/optics.py       186     18    90%
pymatgen/io/vasp/outputs.py     2350    211    91%
pymatgen/io/vasp/sets.py        1204     65    95%
--------------------------------------------------
TOTAL                           4849    438    91%
```

File: .coverage.ZYZhu-M40.dhcp.lbl.gov.36776.XWbBpJZx
Comment (janosh) [inline .coverage.ZYZhu-M40.dhcp.lbl.gov.36776.XWbBpJZx:1]: general hint: no need to check coverage files into version control.

**Review by janosh — APPROVED** [review]
thanks a lot @Zhuoying! 👍 i feel safer already with this new test coverage (no joke) 😄

**Comment by JaGeo** [comment]
Yes, thanks a lot for these efforts!

**Comment by utf** [comment]
This is great, thanks very much @Zhuoying

File: .coverage.ZYZhu-M40.dhcp.lbl.gov.36776.XWbBpJZx
Comment (Zhuoying) [reply .coverage.ZYZhu-M40.dhcp.lbl.gov.36776.XWbBpJZx:1]: Thanks for the hint. I guess I did it unconsciously...

**Pull request merged**